### ISSUE TITLE
Implementation requires PwmPin<Duty=u16> but any Duty value would work

### ISSUE THREAD

**Issue by jasta** [rursprung/tb6612fng-rs#8]
We can generalize further by only changing a few lines of code I believe:

```rust
let max_duty = self.pwm.get_max_duty();
let duty_f = (f64::from(speed) / 100.0) * f64::from(max_duty);
let duty = duty_f.round() as PWM::Duty;
```

**Comment by rursprung** [comment]
i remember ending in generics hell when i tried this (but also didn't invest a lot of time, either). but if you get it to work feel free to provide a PR!

please note that with e-h 1.0 this will be a lot cleaner, see https://github.com/rust-embedded/embedded-hal/pull/430
once there's a new alpha release of e-h 1.0 which includes the new PWM APIs i plan to add support for it here

**Comment by rursprung** [comment]
@jasta: i just released v0.2.0 which is now based on `embedded-hal=1.0.0-rc.1` (see #25 for details). in this version the data type for `Duty` is gone due to the new PWM-related APIs in e-h 1.0
